client/shaders/shadow_shaders/pass2/opengl_fragment.glsl

```diff
@@ -4,18 +4,24 @@ uniform sampler2D ShadowMapClientMapTraslucent;
 #endif
 uniform sampler2D ShadowMapSamplerdynamic;
 
+#ifdef GL_ES
+varying mediump vec2 varTexCoord;
+#else
+centroid varying vec2 varTexCoord;
+#endif
+
 void main() {
 
 #ifdef COLORED_SHADOWS
-	vec2 first_depth = texture2D(ShadowMapClientMap, gl_TexCoord[0].st).rg;
-	vec2 depth_splitdynamics = vec2(texture2D(ShadowMapSamplerdynamic, gl_TexCoord[2].st).r, 0.0);
+	vec2 first_depth = texture2D(ShadowMapClientMap,varTexCoord).rg;
```

Comment:
Why did we have three different texcoords before? I see in the old pass2_vertex shader we set them all to the same value, was there a point to that?

Reply:
My guesses are:
* either it was a constraint when this feature was implemented the first time to have them different, and then they got aligned before the feature got merged. In short: looks like technical debt.
* either it was to let the future possibility to have different UV for the different textures, but I don't see a reason to do so, usually models have 1 UV unwrapping, and the textures conform to it (and not the opposite way, it's rarely the UV that adapts itself to fit the texture).

This file and associated vertex shader only have one commit, dated of 2021, it sounds old enough to say the future this might have been planned for did not happen